otter/log/spec.py

```diff
@@ -0,0 +1,91 @@
+"""
+Format logs based on specification
+"""
+
+from twisted.python.failure import Failure
+
+
+# mapping from msg type -> message
+msg_types = {
+    "execute-convergence": "Executing convergence",
+    "execute-convergence-results": (
+        "Got result of {worst_status} after executing convergence"),
+    "launch-servers": "Launching {num_servers} servers",
+    "delete-server": "Deleting {server_id} server",
+    "add-server-clb": ("Adding {server_id} with IP address {ip_address} "
+                       "to CLB {clb_id}"),
+    "remove-server-clb": ("Removing server {server_id} with IP address "
+                          "{ip_address} from CLB {clb_id}"),
+}
+
+
+def error_event(event, failure, why):
+    """
+    Convert event to error with failure and why
+    """
+    event["isError"] = True
+    event["failure"] = failure
+    event["why"] = why
+    event["original_message"] = event["message"]
+    event["message"] = ()
+    return event
+
+
+def SpecificationObserverWrapper(observer):
+    """
+    Return observer that validates messages based on specification
+    and delegates to given observer
+    """
+    def validating_observer(event_dict):
+        try:
+            speced_event = get_validated_event(event_dict)
+        except (ValueError, TypeError):
+            speced_event = error_event(
+                event_dict, Failure(), "Error validating event")
+        observer(speced_event)
+
+    return validating_observer
+
+
+def try_msg_type(msg_type):
+    """
+    Try message type and return None if not found
+    """
+    return msg_type is not None and msg_types.get(msg_type, None)
```

Comment:
This seems like it'd return `False` if `msg_type` is `None`?  The logic in `get_validated_event` seems to be expecting `None` or a string, though.